### ISSUE TITLE
Discrepancies between onnx and sklearn probabilities with LogisticRegression

### ISSUE THREAD

**Issue by vassilismin** [onnx/sklearn-onnx#1129]
In the following script I'm training a LogisticRegression model into a small classification dataset and i convert it into onnx. Then i use both sklearn and onnx runtimes to compare predicted classes and estimated probabilities. Although the predicted labels are identical between the two runtimes, i noticed significant discrepancies between the estimated probabilities. Is there any fix about this or anyone else had the same issue? 

It is important to note that these differences are not noticed in other classifirers, such as RandomForestClassifier or MLPClassifier. The discrepancies systematically appear when I use LogisticRegression.

Any comment or suggestion will be much appreciated!

Here is my code:
```
import numpy as np
import pandas as pd
from sklearn.linear_model import LogisticRegression
import skl2onnx
from onnxruntime import InferenceSession

# Create a small dataframe with 10 rows and 2 columns
np.random.seed(0)
data = {
    "float_column": np.random.rand(10).astype(np.float64),
    "int_column": np.random.randint(0, 100, size=10).astype(np.int64),
}
x = pd.DataFrame(data)
y = np.random.binomial(1, 0.5, size=10)

# Create a test dataset with 10 rows
test_data = {
    "float_column": np.random.rand(10).astype(np.float64),
    "int_column": np.random.randint(0, 100, size=10).astype(np.int64),
}
x_test = pd.DataFrame(test_data)

# Select and train a model
model = LogisticRegression()
model.fit(x, y)
# Take predictions and probabilities with sklearn
sklearn_preds = model.predict(x_test)
sklearn_probs = model.predict_proba(x_test)

# Convert the model to ONNX
onnx_model = skl2onnx.to_onnx(model, x.values, options={"zipmap": False})
# Take predictions and probabilities with ONNX
sess = InferenceSession(onnx_model.SerializeToString())
onnx_prediction = sess.run(None, {"X": x_test.to_numpy()})

print("SKLEARN Predicted Classes: ", sklearn_preds)
print("ONNX Predicted Classes: ", onnx_prediction[0])
print("SKLEARN Predicted Probabilities: ", sklearn_probs)
print("ONNX Predicted Probabilities: ", onnx_prediction[1])
```

Below are the printed results:

```
SKLEARN Predicted Classes:  [1 1 1 1 1 1 1 1 1 1]
ONNX Predicted Classes:  [1 1 1 1 1 1 1 1 1 1]
SKLEARN Predicted Probabilities:  [[0.06029931 0.93970069]
 [0.16476416 0.83523584]
 [0.08698455 0.91301545]
 [0.44961074 0.55038926]
 [0.05881291 0.94118709]
 [0.18793501 0.81206499]
 [0.27727221 0.72272779]
 [0.13718404 0.86281596]
 [0.05460361 0.94539639]
 [0.33352453 0.66647547]]
ONNX Predicted Probabilities:  [[0.00410073 0.99589927]
 [0.03745654 0.96254346]
 [0.00899505 0.99100495]
 [0.40023472 0.59976528]
 [0.00388956 0.99611044]
 [0.05083641 0.94916359]
 [0.12830092 0.87169908]
 [0.02465633 0.97534367]
 [0.00332482 0.99667518]
 [0.20027542 0.79972458]]
```

**Comment by xadupre** [comment]
I could not replicate your bug, see PR #1131. Maybe this happens which an earlier version. The CI should find let us know. Feel free to comment the PR if you think I tested it in a wrong way.

**Comment by xadupre** [comment]
CI does not fail. What are the versions you are using?

**Comment by vassilismin** [comment]
@xadupre  first of all thank you for your time and response.

I run locally the pr test you created and it still fails for Logistic regression. 

The versions i'm using in my venv are:
python=3.10.14
skl2onnx=1.16.0
onnxruntime=1.18.0

**Comment by vassilismin** [comment]
I updated to the latest vesrions of skl2onnx and onnxruntime and the latest versions and the results are perfect without any discrepancy. Thank you for your help!

**Comment by vassilismin** [comment]
I updated to the latest vesrions of skl2onnx and onnxruntime and the latest versions and the results are perfect without any discrepancy. Thank you for your help!
